smolyay/smolyak.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+import scipy.special
+
+
+class IndexGrid:
+    """Generate index of grid points.
+
+    Attributes
+    ----------
+    dimension : int
+        Number of independent variables.
+    exactness : int
+        Level of exactness of the approximation.
+    index_per_level : list
+        Number of indexes at each level.
+    """
+
+    def __init__(self, dimension, exactness, index_per_level):
+        self.dimension = dimension
+        self.exactness = exactness
+        self.index_per_level = index_per_level
+
+        if len(index_per_level) < exactness + 1:
+            raise IndexError(
+                "index_per_level must be an array with a"
+                "length of at least {}"
+                .format(exactness + 1))
+
+    def level_indexes(self):
+        """Generate indexes of levels.
+
+        Returns
+        -------
+        level_indexes : list
+            Unique indexes at each level.
+
+        """
+        # creating an array to store unique indexes in each level
+        level_indexes = [[0]]*(self.exactness + 1)
+        max_index = sum(self.index_per_level[0:self.exactness + 2])
+        indexes = np.linspace(0, max_index, max_index+1, dtype=np.int32)
+
+        # intial level
+        level_indexes[0] = indexes[0:self.index_per_level[0]]
+
+        # storing unique indexes for each level
+        summ_index = self.index_per_level[0]
+        for idx_index_per_level in range(1, self.exactness + 1):
+            summ_index += self.index_per_level[idx_index_per_level]
+            level_indexes[idx_index_per_level] = (
+                indexes[summ_index-self.index_per_level[idx_index_per_level]:
+                        summ_index])
+
+        return level_indexes
+
+    def grid_point_index(self):
```

Comment:
This will also become a property, so it should get a short docstring like above. The comments about NEXCOM, etc. belong up in the detailed class description, which still needs to be written.

Reply:
Done.